### PR TITLE
Refactor/check scheduler

### PR DESCRIPTION
**Pull request by echlebek**
## What is this change?

This branch substantially reworks schedulerd, removing the caching layer in favour of direct accesses to etcd. This should resolve bugs related to caching in the scheduler, as well as resolve ambiguity about what the current state of the scheduler is.

## Why is this change necessary?

The scheduler is difficult to reason about, and the code is poorly factored.

## Does your change need a Changelog entry?

Yes

### CODE REVIEW

File: backend/schedulerd/executor.go [hunk 1]
```diff
@@ -233,7 +238,7 @@ func (a *AdhocRequestExecutor) publishProxyCheckRequests(entities []*types.Entit
 }
 
 func (a *AdhocRequestExecutor) execute(check *types.CheckConfig) error {
-	request := a.buildRequest(check)
+	request, _ := a.buildRequest(check)
```
Comment (nikkiki) [inline backend/schedulerd/executor.go]: Would it be beneficial to catch and return the error here from `buildRequest`?

File: util/time/time.go [hunk 2]
```diff
@@ -1,105 +0,0 @@
...
-	}
-
-	return current.After(beginTime) && current.Before(endTime), nil
-}
-
-// InWindows determines if the current time falls between the provided time
-// windows. Current should typically be time.Now() but to allow easier tests, it
-// must be provided as a parameter. The function returns a positive value as
-// soon the current time falls within a time window
-func InWindows(current time.Time, timeWindow types.TimeWindowWhen) (bool, error) {
```
Comment (nikkiki) [inline util/time/time.go]: Was this a duplicate function? You still invoke `subdue.InWindows` but this PR appears to remove, not replace, the function, as it exists in the `types` package already.

File: backend/schedulerd/executor.go (hunk 1 above)
Comment (echlebek) [reply backend/schedulerd/executor.go]: In this instance, nil is always returned by AdhocRequestExecutor.buildRequest.

File: util/time/time.go (hunk 2 above)
Comment (echlebek) [reply util/time/time.go]: I moved this code over some time ago, but did not quite get around to removing all existing uses of it. Now, it is only called from the types package.

Comment (nikkiki) [reply util/time/time.go]: Got it.. thanks for the extra cleanup!

File: backend/schedulerd/executor.go (hunk 1 above)
Comment (nikkiki) [reply backend/schedulerd/executor.go]: Hopefully if that changes in the future, we remember to catch it.. I don't foresee that happening though 😉

**Comment by echlebek** [comment]
Closes #1573

**Pull request merged**